### DISCUSSION TITLE
[ZK Stack] - <ways to deploy multi-provers in different machines>

### DISCUSSION THREAD

**Discussion by qiwenzhou** [zkSync-Community-Hub/zksync-developers#604] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [X] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.74.0

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

What is the proper way to deploy multi-prover for my server. I ran a server and a prover on one machine A , worked fine.
After that, I tried to add one more prover by: export DATABASE_PROVER_URL=postgres://postgres:notsecurepassword@A_IP/prover_local on machine B and ran zk f cargo run --features "gpu" --release --bin zksync_prover_fri.
At the beginning, the two prover worked fine.Both prover could get jobs. After certain time, there always be a problem to come up.
![img_v3_02bv_d03f6f12-b8d2-4867-987e-12292811c2hu](https://github.com/zkSync-Community-Hub/zksync-developers/assets/31309283/a9792f5b-f60e-466d-aae2-4e3d6e2020f6)
It seems like zksync_witness_generator failed to process "aggregation type" jobs due to lack of files. More than that, zksync_witness_generator stopped to produce prove jobs for new batches.

### Expected Behavior

Run multi-provers on multi-machines with one prover on one machine . How to achieve that?

### Current Behavior

will fail.

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by bxpana** [comment]
Not sure if we have documentation about this quite yet, but will check on this for you.

**Reply by qiwenzhou** [reply]
Thank you!

**Comment by qiwenzhou** [comment]
UPDATE:
I turned down postgres in machine B,pointing prover database to machine A and prover on machine B failed to verify proofs after some successful jobs.
![img_v3_02bv_0c7ca19d-49de-4c1c-8883-3ec92289cdhu](https://github.com/zkSync-Community-Hub/zksync-developers/assets/31309283/7dc86752-a9a4-4681-ad4a-1c8211ccd27e)

**Reply by qiwenzhou** [reply, marked as answer]
FIXED：It turned out that provers in two machine store the data in defferernt positions. After making prover in machine B to fetch and store data in machine A, it worked.
